acceptance/acceptance_test.go

```diff
@@ -704,6 +704,64 @@ func testAcceptance(
 				builderName = value
 			})
 
+			when("complex builder", func() {
+				it.Before(func() {
+					// create our nested builder
+					h.SkipIf(t, imageManager.HostOS() == "windows", "These tests are not yet compatible with Windows-based containers")
+					h.SkipIf(t, !createBuilderPack.SupportsFeature(invoke.BuilderNoDuplicateLayers), "bug fixed in 0.18.0")
+
+					// create a task, handled by a 'task manager' which executes our pack commands during tests.
+					// looks like this is used to de-dup tasks
+					key := taskKey(
+						"create-complex-builder",
+						append(
+							[]string{runImageMirror, createBuilderPackConfig.Path(), lifecycle.Identifier()},
+							createBuilderPackConfig.FixturePaths()...,
+						)...,
+					)
+					// run task on taskmanager and save output, in case there are future calls to the same task
+					// likely all our changes need to go on the createBuilderPack.
+					value, err := suiteManager.RunTaskOnceString(key, func() (string, error) {
+						return createComplexBuilder(
+							t,
+							assert,
+							createBuilderPack,
+							lifecycle,
+							buildpackManager,
+							runImageMirror,
+						)
+					})
+					assert.Nil(err)
+
+					// register task to be run to 'clean up' a task
+					suiteManager.RegisterCleanUp("clean-"+key, func() error {
+						imageManager.CleanupImages(value)
+						return nil
+					})
+					builderName = value
+
+					output := pack.RunSuccessfully(
+						"config", "run-image-mirrors", "add", "pack-test/run", "--mirror", "some-registry.com/pack-test/run1")
+					assertOutput := assertions.NewOutputAssertionManager(t, output)
+					assertOutput.ReportsSuccesfulRunImageMirrorsAdd("pack-test/run", "some-registry.com/pack-test/run1")
+				})
+				when("builder has duplicate buildpacks", func() {
+					it("buildpack layers have no duplication", func() {
+						out, _, err := dockerCli.ImageInspectWithRaw(context.Background(), builderName)
+						assert.Nil(err)
+
+						layerSet := map[string]interface{}{}
+						for _, layer := range out.RootFS.Layers {
+							_, ok := layerSet[layer]
+							if ok {
+								t.Fatalf("duplicate layer found in builder %s", layer)
+							}
+							layerSet[layer] = true
+						}
```

Comment:
Can this be moved into the `ImageAssertionManager` (https://github.com/buildpacks/pack/blob/main/acceptance/assertions/image.go#L13-L18)? It could be useful for reuse and for clarity